app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```
@@ -0,0 +1,138 @@
+      ...defaultState,
+      ...initState,
+    });
+    this.metaMetricsController = metaMetricsController;
+  }
+
+  _formatDeletionDate() {
+    const currentDate = new Date();
+    const day = currentDate.getUTCDate();
+    const month = currentDate.getUTCMonth() + 1;
+    const year = currentDate.getUTCFullYear();
+
+    // format the date in the format DD/MM/YYYY
+    return `${day}/${month}/${year}`;
+  }
+
+  async createMetaMetricsDataDeletionTask(): Promise<DataDeletionResponse> {
+    const segmentSourceId = process.env.SEGMENT_DELETE_API_SOURCE_ID;
+    const segmentRegulationEndpoint = process.env.SEGMENT_REGULATIONS_ENDPOINT;
+
+    if (!segmentSourceId || !segmentRegulationEndpoint) {
+      return {
+        status: responseStatus.error,
+        error: 'Segment API source ID or endpoint not found',
+      };
+    }
+    const { metaMetricsId } = this.metaMetricsController.store.getState();
+
+    try {
+      const { data } = await createDataDeletionRegulationTask(
+        metaMetricsId as string,
+      );
+
+      this.store.updateState({
+        metaMetricsDataDeletionId: data?.data?.regulateId as string,
+        metaMetricsDataDeletionDate: this._formatDeletionDate(),
+      });
+
+      return { status: responseStatus.ok };
+    } catch (error: unknown) {
+      return {
+        status: responseStatus.error,
+        error: error as string,
+      };
+    }
+  }
+
+  async checkDataDeletionTaskStatus(): Promise<DataDeletionResponse> {
+    const segmentRegulationEndpoint = process.env.SEGMENT_REGULATIONS_ENDPOINT;
+
+    if (!segmentRegulationEndpoint) {
+      return {
+        status: responseStatus.error,
+        error: 'Segment API source ID or endpoint not found',
+      };
+    }
+    try {
+      const { data } = await fetchDeletionRegulationStatus(
+        this.store.getState().metaMetricsDataDeletionId as string,
+      );
+      const regulation = data?.data?.regulation as Record<string, string>;
```

If `data?.data?.regulation` is `undefined`, should we return an error?

---

I believe the api will return a regulation if we have a valid Id, otherwise, it would probably error. I have added an empty check for `deleteRegulateId` to make sure we have a value before we call the api..